### DISCUSSION TITLE
Does it work with a express and Prismic configuration in a webpack.config file ?

### DISCUSSION THREAD

**Discussion by orpheebesson** [ReeceM/prismic-sitemap#17] [Q&A] [answered]
Hello,

I have the following configuration in my project :

- `app.js` which is my router for my different url's
- `webpack.config.ts` which is my webpack configuration file

I would like to know if it is possible to generate a sitemap with this configuration, or it only works with a `next.config.js`.

Thanks in advance.

[prismic-sitemap-q&a.zip](https://github.com/ReeceM/prismic-sitemap/files/9484547/prismic-sitemap-q.a.zip)

**Comment by ReeceM** [comment]
Hi @orpheebesson 

I actually haven't tried to run it through an express app. 

But there is a file called generator.js. It's not directly exported, but you should be able to access it directly. 

You can pass it the same config that is sent to the next.config part.

**Reply by ReeceM** [reply]
Also you would pass the object as `{...configOptions}` not `{sitemap: {...configOptions}}`

**Reply by orpheebesson** [reply]
Thank for your answer @ReeceM,

so if I understand, I can use the generator.js file with the next.config configuration right ?

I add the code in my `app.js` file or rather in my `webpack.config.ts` ? Because prismic-sitemap is not a Webpack Plugin.

If you can enlighten me on this...

**Reply by ReeceM** [reply]
Yes, you can import/require straight into the express app with the same config settings you would have used. So you are right 👍 

It isn't a Webpack plugin, but extends the sitemap javascript library and includes specifics for Prismic like auto pagination and such :)

**Reply by ReeceM** [reply]
So you would do like so:

```javascript
const generator = require("@reecem/prismic-sitemap/generator.js');

// example from the next config
await generator({
        linkResolver: linkResolver,
        apiEndpoint: API_ENDPOINT,
        hostname: SITE_URL,
        optionsMapPerDocumentType: {
          // setting the update date of the article.
          post: (document) => {
            return { 
              // get the last time the document was published in Prismic
              lastmod: document.last_publication_date,
              changefreq: "weekly", 
              priority: 0.8 
            }
          }
          page: { changefreq: "monthly", priority: 1 }
        },
        documentTypes: ['page', 'post']
    });

```

**Reply by orpheebesson** [reply]
Thank fou for your example.

I tried this code below but no sitemap was generated, if you have an idea please :

```
const generator = require('@reecem/prismic-sitemap/generator')

const buildSitemap = async () => {
  const sitemap = await generator({
    linkResolver: handleLinkResolver,
    apiEndpoint: process.env.PRISMIC_ENDPOINT,
    accessToken: process.env.PRISMIC_ACCESS_TOKEN,
    hostname: process.env.SITE_URL,
    optionsMapPerDocumentType: {
      post: (document) => {
        return {
          lastmod: document.last_publication_date,
          changefreq: "weekly",
          priority: 0.8
        }
      },
      page: { changefreq: "monthly", priority: 1 }
    },
    documentTypes: ['post', 'page', 'project']
  })

  return sitemap
}

buildSitemap()
```

**Reply by ReeceM** [reply]
Let me have a look at it in an express app then, as I have the similar function call to the generator in a normal JS file. Will get back to you on that

**Comment by ReeceM** [comment, marked as answer]
Hey @orpheebesson so if you set the config option in the generator `publicPath: path.join(__dirname, 'public'),` it should create the file. You will actually find the one it's been generating in `node_modules/@reecem/prismic-sitemap/public/sitemap.xml`

This is because the index.js file in the package determines the dir to place it in if it isn't specified. And then it calls the generator. Forgot about that :) But should work after that

**Reply by orpheebesson** [reply]
Thank you very much for your investigation. Sitemap generation works perfectly now !

Thanks again @ReeceM

**Reply by ReeceM** [reply]
> Thank you very much for your investigation. Sitemap generation works perfectly now !
> 
> 
> 
> Thanks again @ReeceM 

Pleasure, glad it is working and I think it may be something then to add to the README that it's possible 😁
